test/cppsuite/test_harness/workload/database_operation.h

```diff
@@ -144,10 +144,16 @@ class database_operation {
 
         while (tc->running()) {
             /* Walk each cursor. */
+            if (!tc->transaction.active())
```

Comment:
I realised we should put these readers into transactions to avoid pinning content for too long.